sulfnucrate.F90

```diff
@@ -4,51 +4,386 @@
   use carma_types_mod
   use carmastate_mod
   use carma_mod
   use sulfate_utils
-  
+
   implicit none
-  
-  type(carma_type), intent(in)         :: carma       !! the carma object
-  type(carmastate_type), intent(inout) :: cstate      !! the carma state object
-  integer, intent(in)                  :: iz          !! level index
-  integer, intent(in)                  :: igroup      !! group index
-  real(kind=f), intent(out)            :: h2o         !! H2O concentrations in molec/cm3 
-  real(kind=f), intent(out)            :: h2so4       !! H2SO4 concentrations in molec/cm3
+
+  type(carma_type), intent(in)         :: carma          !! the carma object
+  type(carmastate_type), intent(inout) :: cstate         !! the carma state object
+  integer, intent(in)                  :: iz             !! level index
+  integer, intent(in)                  :: igroup         !! group index
+  real(kind=f), intent(out)            :: h2o            !! H2O concentrations in molec/cm3
+  real(kind=f), intent(out)            :: h2so4          !! H2SO4 concentrations in molec/cm3
   real(kind=f), intent(out)            :: beta1
   real(kind=f), intent(out)            :: beta2
   real(kind=f), intent(out)            :: ftry
-  real(kind=f), intent(out)            :: rstar       !! critical radius (cm)
-  integer, intent(out)                 :: nucbin      !! bin in which nucleation occurs
-  real(kind=f), intent(out)            :: nucrate     !! nucleation rate #/x/y/z/s
-  integer, intent(inout)               :: rc          !! return code, negative indicates failure
+  real(kind=f), intent(out)            :: radius_cluster !! critical radius (cm)
+  integer, intent(out)                 :: nucbin         !! bin in which nucleation occurs
+  real(kind=f), intent(out)            :: nucrate        !! nucleation rate #/x/y/z/s
+  integer, intent(inout)               :: rc             !! return code, negative indicates failure
+
+  !  Local declarations
+  integer           :: i, ibin, ie
+  real(kind=f)      :: nucrate_cgs      ! binary nucleation rate, j (# cm-3 s-1)
+  real(kind=f)      :: cnum_h2so4       ! number of h2so4 molecules in the critical nucleus
+  real(kind=f)      :: cnum_tot         ! total number of molecules in the critical nucleus
+  real(kind=f)      :: rb               ! [erg/mol]
+  real(kind=f)      :: h2so4_cgs        ! H2SO4 densities in g/cm3
+  real(kind=f)      :: h2o_cgs          ! H2O densities in g/cm3
+  real(kind=f)      :: rh               ! relative humidity (0-1)
+  real(kind=f)      :: mass_cluster_dry ! dry mass of the cluster ()
+  real(kind=f)      :: h2so4_bb         ! bounded value of H2SO4 concentration in molec/cm3
+  real(kind=f)      :: temp_bb          ! bounded value of temperature in Kelvins
+  real(kind=f)      :: rh_bb            ! bounded value of relative humidity
+
+  integer, parameter :: newnuc_method_flagaa = 1
+  !  1=Zhao & Turco (1995) binary
+  !  2=vehkamaki et al (2002) binary
+
```

Comment:
remove these lines

Reply:
These lines shouldn't been in here. Also in CARMA all the enumerations are in carma_enums_mod.F90, and typically have been numbers not strings. I don't really care if they are numbers or strings, but they should be consistent to make it easier to read the code and find things.